packages/mui-codemod/src/deprecations/popper-props/popper-props.js

```diff
@@ -0,0 +1,24 @@
+import { EOL } from 'os';
+
+import replaceComponentsWithSlots from '../utils/replaceComponentsWithSlots';
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const printOptions = options.printOptions;
+
+  replaceComponentsWithSlots(j, { root, componentName: 'Popper' });
+
+  // I've tried { lineTerminator } option here, but still get wrong line endings on my Windows machine
+  const result = root.toSource(printOptions);
+
+  if (EOL !== '\n') {
+    return result.replace(/\n/g, EOL);
+  }
```

Comment:
What happens without this? Do you experience the same with other codemods?

I'm wondering if we need to do something similar on all codemods.

@sai6855 what OS are you running? Have you experienced this?

Reply:
I ran on both mac and windows, didn't faced any issues

Reply:
Hello. I am running the code on my Windows machine and found out that tests are fallen due to line endings. You can take a look on that PR: https://github.com/mui/material-ui/pull/42093 I described the issue there. Also in that PR I moved out `transformer` function in utils. It will help to reuse the same code for transformers.
So to be much detailed:
In transformer function `root.toSource()` on Windows machine returns string with `\n` line endings. Seems like the issue is in the `recast` library that is used by `jscodeshift` one.
For `readFile` we have:
```
if (EOL !== '\n') {
  return fileContents.replace(/\n/g, EOL);
}
```
Windows machine has `\r\n` as EOL and this code always returns for me `\r\n` line endings.
That is why I need to convert `\n` to `\r\n` because I have my own transformer function and `readFile` is in utils and reused by other tests that can break something other components tests

Reply:
As you mentioned in your comment, recast has the option to configure the line terminator: https://github.com/facebook/jscodeshift/issues/195. Does this fix your issue? ([how to provide recast options](https://github.com/mui/material-ui/tree/next/packages/mui-codemod#recast-options))

These formatting options are handled by recast, not us. If there's something not working with formatting, then the fix should go there as well.

To move forward, we have to remove this `EOL` patch. 

Does that make sense?

Reply:
Updated